project/MyScalaNativePlugin.scala

```diff
@@ -55,6 +67,83 @@ object MyScalaNativePlugin extends AutoPlugin {
     }
   }
 
+  private def nativeLinkProfilingImpl = Def.inputTaskDyn {
+    val sbtLogger = streams.value.log
+    val logger = sbtLogger.toLogger
+
+    val args = spaceDelimited("<arg>").parsed
+
+    val commands = args.headOption match {
+      case None =>
+        throw new IllegalArgumentException(
+          "usage: nativeLinkProfiling <commands> <output-type>\n" +
+            "<commands>: `,` delimited arguments for async-profiler. refer https://github.com/async-profiler/async-profiler/blob/49d08fd068f81f1c952320c4bd082d991e09db97/src/arguments.cpp#L65-L113 \n" +
+            "<output-type>: text|collapsed|flamegraph|tree (default: flamegraph) \n" +
+            "e.g. `nativeLinkProfiling events=cpu,interval=10000000,threads"
+        )
+      case Some(value) =>
+        value
+    }
+
+    val outputType = (for {
+      input <- args.tail.headOption.toRight(
+        new IllegalArgumentException("Missing output type")
+      )
+      typ <- OutputType.fromString(input)
+    } yield typ) match {
+      case Left(ex) =>
+        logger.warn(
+          s"${ex.getMessage()}, using default output type: `flamegraph`"
+        )
+        OutputType.Flamegraph
+      case Right(value) => value
+    }
+
+    val profilerOpt: Option[AsyncProfiler] =
+      if (AsyncProfilerLoader.isSupported())
+        try {
+          Some(AsyncProfilerLoader.load())
+        } catch {
+          case ex: IllegalStateException => {
+            logger.warn(
+              s"Couldn't load async-prfiler, restart sbt to workaround the problem. " +
+                "This is usually caused because old sbt's classloader loaded the async-profiler DLL. \n" +
```

Comment:
```suggestion
              s"Couldn't load async-profiler, restart sbt to workaround the problem. " +
                "This is usually caused because the old sbt's classloader loaded the async-profiler DLL. \n" +
```

Reply:
Maybe the `previous` sbt classloader?